### ISSUE TITLE
Support for export import statements

### ISSUE THREAD

**Issue by fede-s** [TypeStrong/typedoc#159]
I have a library that uses requirejs modules and then is published as a node module.
I like to structure my project so that there is only 1 class per file, and then I have 1 file that is the library entry point file that exports all the classes I want to make available.
See the following example

file lib-entry-point.ts

```
export import ClassA = require('./ClassA');
//Here I change the name of ClassB and would like that it's name on the docs is the name I choose here
export import IChangedTheName = require('./ClassB');
export import module1 = require('./module1');

/**
 * does something
 */
export function aFunction () {

}
```

file ClassA.ts

```
/**
 * Class named A
 */
class ClassA  {
}

export = ClassA;

```

file ClassB.ts

```
/**
 * Class named B
 */
class ClassB  {
    public name = "B";
}

export = ClassB;
```

file module1.ts

```
/**
 * module that exports Class C
 */
export import ClassC = require('./ClassC');
```

file ClassC.ts

```
`/**
 * Class named C
 */
class ClassC  {
}

export = ClassC;
```

To generate the documentation I chose "file" mode and added the 5 files to the src property. What I get is:

![screen shot 2015-12-01 at 12 17 24](https://cloud.githubusercontent.com/assets/5686196/11487674/b97790a8-9825-11e5-9d56-4898e58ccbb2.png)

But, I would like to get this:

![screen shot 2015-12-01 at 12 22 26](https://cloud.githubusercontent.com/assets/5686196/11487762/7408642e-9826-11e5-9b8f-98f288aadcde.png)

The problem is that if I want to get that result, ClassB name must be "IChangedTheName" and ClassC has to be contained within module1.ts

```
/**
 * module that exports Class C
 */
module module1 {
    /**
     * Class named C
     */
    export class ClassC  {
    }
}

export = module1;

```

So my question is, would it be possible to have typedoc interpreting the structure of the project from the export import statements and be able to set whatever name I want for my class at that point?

**Comment by sebastian-lenz** [comment]
Hi, thank you for your long comment. I see your point here and it makes sense to me. I have to admit that TypeDoc currently has absolutely no special handling for import/export statements and I also see the need for something like that. I'm still quite unsure on how to implement something like this as there are various use cases of import/export statements. My first step would have been to add a basic export table to the documentation al least gives basic information about the exported members of a module.

**Comment by iRath96** [comment]
I've had this issue in a library of mine recently and decided to fork typedoc to implement this feature.
My implementation is by no means complete (and not very elegant either) but it might be useful nonetheless :)

https://github.com/iRath96/typedoc

**Comment by TheNeuralBit** [comment]
Is there any interest in closing this out by syncing and merging either https://github.com/iRath96/typedoc or #646? We export imports extensively in Apache Arrow and it would be great to have better support for this.

See [ARROW-951](https://issues.apache.org/jira/browse/ARROW-951?focusedCommentId=16595187&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-16595187) for more on our particular issue

**Comment by Gerrit0** [comment]
It would be great to merge in #646 (looks more complete than iRath96's implementation) - we currently have quite a backlog of PRs and issues to address that are simply waiting on time.

**Comment by TheNeuralBit** [comment]
Understood. Anything I can do to help move #646 along? I could look into syncing it up with master if you think that would help.

**Comment by Gerrit0** [comment]
That would be helpful since it would make it easier to pull and check that everything is working correctly before merging. Thanks!

**Comment by Gerrit0** [comment]
Support for this has been merged with #1157
